### PR TITLE
Add a config switch to require code owner reviews

### PR DESCRIPTION
**Pull request by coheigea**
Signed-off-by: Colm O hEigeartaigh <coheigea@apache.org>

Implemented https://github.com/ossf/allstar/issues/312

### CODE REVIEW

File: pkg/policies/branch/branch.go [hunk 1]
```diff
@@ -566,6 +582,10 @@ func fix(ctx context.Context, rep repositories, c *github.Client,
 				pr.RequiredPullRequestReviews.RequiredApprovingReviewCount = mc.ApprovalCount
 				update = true
 			}
+			if mc.RequireCodeOwnerReviews && !pr.RequiredPullRequestReviews.RequireCodeOwnerReviews {
```
Comment (jeffmendoza) [inline pkg/policies/branch/branch.go:586]: Should L576 above also change to `if mc.RequireApproval || mc.RequireCodeOwnerReviews `?

Comment (jeffmendoza) [reply pkg/policies/branch/branch.go:586]: Also L567

**Review by jeffmendoza — CHANGES_REQUESTED** [review]
Looks great, thanks. A couple questions about the if statements.
Alternatively, The help for the config (L55) could stipulate that that `requireCodeOwnerReviews` only applies if `requireApproval` is true. This might simplify the logic a bit.

File: pkg/policies/branch/branch.go (hunk 1 above)
Comment (coheigea) [reply pkg/policies/branch/branch.go:586]: I've updated the help, as I think it doesn't make sense to use requireCodeOwnerReviews without requireApproval

**Review by jeffmendoza — APPROVED** [review]

**Comment by jeffmendoza** [comment]
Thank you!

I'll do a little manual testing in staging before pushing to prod.
Do you have anything else in the works? I can do a release tomorrow if not.

**Comment by coheigea** [comment]
No nothing else planned, thanks.

**Pull request merged**